### ISSUE TITLE
How to implement Convolution2D with filter size of 3x3 and stride 2x2 ?

### ISSUE THREAD

**Issue by jingzhehu** [fchollet/keras#2434]
I was trying to implement a 2d convolution layer with filter size of 3x3x(32) and stride 2x2 using TensorFlow backend. If the input size is (299x229x3), I would expect the output size to be (149x149x32) with valid padding. [output_size = (229-3+0)/2+1 = 149]

However, after setting **subsample** to (2,2), I get an output size of (114 x 114 x 32) instead. Not quite sure what the problem is with my code and I would really appreciate any help or pointers ! 

```python
from keras.layers import Input, Convolution2D, MaxPooling2D, Flatten
from keras.utils.visualize_util import plot

inputs = Input(shape=(229, 229, 3))

x = Convolution2D(32, 3, 3, subsample=(2, 2), border_mode='valid', dim_ordering='tf')(inputs)`

x = Flatten()(x)
loss = Dense(32, activation='relu', name='loss')(x)
model = Model(input=inputs, output=loss)
model.compile(optimizer='rmsprop', loss='binary_crossentropy')

# visualize model layout with pydot_ng
plot(model, to_file='model.png', show_shapes=True)

```

Edit 1: I'm also having similar issues with MaxPooling2D.
`MaxPooling2D(pool_size=(3, 3), strides=(2, 2), border_mode='valid', dim_ordering='tf')(inputs)`

**Comment by joelthchao** [comment]
So, what's your input size? input size **299** will result in output size **149** while **229** will result in **114**.

**Comment by jingzhehu** [comment]
Ha, typo on my part. The input size shall be 299. Problem resolved !
